.github/workflows/ci.yml

```diff
@@ -31,7 +31,7 @@ on:
         type: string
       PYTHON_MATRIX:
         description: 'Comma-separated list of Python versions e.g. "3.11","3.12"'
-        default: '"3.9.15", "3.10.8", "3.11.0", "3.12.9", "3.13.2"'
+        default: '"3.11.0", "3.12.9", "3.13.2", "3.14.0"'
```

Comment:
Is there any reason for using these specific Python versions? For quirks, we just use `"3.12", "3.13"` (so the latest patch releases?) and for ZHA, we use some other specific Python versions at the moment: `"3.12.10", "3.13.3"`.

Reference:
- https://github.com/zigpy/zha-device-handlers/blob/dev/.github/workflows/ci.yml
- https://github.com/zigpy/zha/blob/dev/.github/workflows/ci.yml

Both ZHA and quirks also don't support 3.11, so we'll still need to override all Python versions to test against there. So, we might wanna streamline the Python versions we use for this everywhere.

Reply:
We had issues with CI in the past when the versions were unconstrained. Different stages of a workflow seemingly used different patch releases and cache just could not be restored properly. Maybe GitHub fixed something?